File: src/main/java/net/yogstation/yogbot/config/DiscordConfig.java
package net.yogstation.yogbot.config;

import discord4j.common.util.Snowflake;
import org.springframework.stereotype.Component;

@Component
public class DiscordConfig extends ConfigClass {
	
	public DiscordConfig() {
		loadConfig("discord.properties");
	}
	
	
	public String botToken = "No token here";
	
	public String commandPrefix = "!";
	public String serverName = "Yogstation 13";
	
	public boolean useLocalCommands = false;
	
	public String oauthAuthorizeUrl = "https://bab.yogstation.net/auth/authorize";
	public String oauthTokenUrl = "https://bab.yogstation.net/auth/token";
	public String oauthUserInfoUrl = "https://bab.yogstation.net/auth/userinfo";
	public String oauthClientId = "yogstation_yogbot";
	public String oauthClientSecret = "lolno";
	
	public String asayWebhookUrl = "";
	public String msayWebhookUrl = "";
	
	public long mainGuildID = 134720091576205312L;
	public long aoRole = 471634210923216906L;
	public long mentorRole = 505280515322937355L;
	public long jesterRole = 539846767827746817L;
	public long subscriberRole = 213375106888499200L;
	
	public long manualVerifyRole = 768196461313523727L;
	public long byondVerificationRole = 762005208326733845L;
	
	// Punishment Roles
	public long softBanRole = 302093934551629827L;
	public long loreBanRole = 716784185192480798L;
	public long mentorBanRole = 865043884170149928L;
	public long wikibanRole = 792982865989992490L;
	
	public long firstWarningRole = 514066643119505408L;
	public long secondWarningRole = 514066798216347662L;
	public long staffPublicBanRole = 290926097028218884L;
}

File: src/main/java/net/yogstation/yogbot/config/DiscordConfig.kt
package net.yogstation.yogbot.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "yogbot.discord")
class DiscordConfig {

	var botToken: String = "no token here"
	var commandPrefix: String = "!"

	var serverName: String = "Yogstation 13"
	var useLocalCommands: Boolean = false

	var oauthAuthorizeUrl: String = "https://bab.yogstation.net/auth/authorize"

	var oauthTokenUrl: String = "https://bab.yogstation.net/auth/token"
	var oauthUserInfoUrl: String = "https://bab.yogstation.net/auth/userinfo"
	var oauthClientId: String = ""
	var oauthClientSecret: String = ""

	var msayWebhookUrl: String = ""
	var asayWebhookUrl: String = ""
	var ticketWebhookUrl: String = ""

	var mainGuildID: Long = 134720091576205312L
	var aoRole: Long = 471634210923216906L
	var mentorRole: Long = 505280515322937355L
	var jesterRole: Long = 539846767827746817L
	var subscriberRole: Long = 213375106888499200L

	var manualVerifyRole: Long = 768196461313523727L
	var byondVerificationRole: Long = 762005208326733845L

	// Punishment Roles
	var softBanRole: Long = 302093934551629827L
	var loreBanRole: Long = 716784185192480798L
	var mentorBanRole: Long = 865043884170149928L
	var wikibanRole: Long = 792982865989992490L

	var firstWarningRole: Long = 514066643119505408L
	var secondWarningRole: Long = 514066798216347662L
	var staffPublicBanRole: Long = 290926097028218884L
}
